### DISCUSSION TITLE
When is ESLint 9 meant to be released?

### DISCUSSION THREAD

**Discussion by nbouvrette** [eslint/eslint#17468] [General] [answered]
I was just reading at some of the documentation today and found this mention [here](https://eslint.org/docs/latest/use/configure/configuration-files):

<img width="847" alt="image" src="https://github.com/eslint/eslint/assets/6453322/7bf2eafe-0bb1-4ba4-a374-3d5c0ceb019e">

I was curious if there was any plans on when this new config would become the default one? And I was also curious if there were plans to support other file format as well other than `.js` file?

**Comment by Rec0iL99** [comment]
Hi @nbouvrette, you can track #13481. Docs on how to use the new config system can be found [here](https://eslint.org/docs/latest/use/configure/configuration-files-new).

**Reply by nbouvrette** [reply]
Thanks for sharing these. I couldn't spot much info on a potential release date or supporting other file formats.

Does this mean that there is no clear plan on that front right now?

**Reply by Rec0iL99** [reply, marked as answer]
There is only one file format in flat config: `eslint.config.js`. 

As far as I know, we don't have a set date planned for the release of v9.

**Note:** Flat config will become the default config system in ESLint v9. The `.eslintrc.*` system will still be available in the v9 release, but using that will show a deprecation warning.

**Reply by nbouvrette** [reply]
Thanks @Rec0iL99 - was there a reason to push for a .`js` file as the only config file or do you think that in the future other formats like `.json` or `.yaml` might be also supported?

**Reply by Rec0iL99** [reply]
I would recommend checking out our blog posts where you will find a detailed [explanation](https://eslint.org/blog/2022/08/new-config-system-part-1/#multiple-config-file-formats) for this decision.

https://eslint.org/blog/2022/08/new-config-system-part-1/
https://eslint.org/blog/2022/08/new-config-system-part-2/
https://eslint.org/blog/2022/08/new-config-system-part-3/

**Reply by nbouvrette** [reply]
This is really good context - I see the benefits of using a single format for simplicication and also leverage JavaScript to allow dyamic configuration.

Unfortunately I couldn't find any mentions of supporting a `eslint.config.ts` which could enhance the configuration files with native typings - is this something that might be considered in the future?

**Reply by Rec0iL99** [reply]
ESLint core is not designed for TypeScript code. TypeScript is handled by a 3rd party ESLint plugin (`typescript-eslint`). So, I don't think there will be official support for a `eslint.config.ts`. There have been discussions in the past (#12078) to support a `.ts` config, but the proposal couldn't get enough support from the team.

**Reply by the-homeless-god** [reply]
https://github.com/typescript-eslint/typescript-eslint/issues/8211

**Comment by doppelmutzi** [comment]
In a 2023 blog post, December 2024 / January 2024 was targeted as the release date for ESLint v9. Do you have an updated plan for when the release will take place? Will be v9.0.0-beta.1 the last pre-release version?

Many thanks for considering my request.

**Reply by Rec0iL99** [reply]
https://github.com/orgs/eslint/projects/4/views/1
https://eslint.org/blog/2023/11/whats-coming-in-eslint-9.0.0/
